tools/autograd/templates/Functions.cpp

```diff
@@ -902,6 +902,40 @@ Tensor log_softmax_double_backward(const Tensor & grad, const Tensor & grad_outp
   return z * grad_output.sum(dim, true) * ((grad * z).sum(dim, true) - grad);
 }
 
+Tensor binary_cross_entropy_double_backward(const Tensor & grad_output, const Tensor & grad, const Tensor & input, const Tensor & target, const Tensor& weight, int64_t reduction) {
+  auto eps = 1e-12;
+  auto num = input * input - 2 * input * target + target;
+  auto denom = (input + eps) * (input + eps) * (1 - input + eps) * (1 - input + eps);
```

Comment:
Every time to do `input + eps` here, you actually create a new Tensor of the size of `input`, and add element wise `eps` to it.
You would speed this up and use less memory by doing (with different naming):
```cpp
auto input_plus_eps = input + eps;
auto one_minus_input_plus_eps = 1 - input_plus_eps;
auto denom = input_plus_eps.pow(2) * one_minus_input_plus_eps.pow(2);
```